internal/ledger/store/badger.go

```diff
@@ -94,7 +94,47 @@ func (b *BadgerDB) Set(key Key, value []byte) error {
 	return nil
 }
 
+// NewIterator returns new Iterator over the store.
+func (b *BadgerDB) NewIterator(scope Scope) Iterator {
+	bi := badgerIterator{scope: scope, fullPrefix: scope.Bytes()}
+	bi.txn = b.backend.NewTransaction(false)
+	opts := badger.DefaultIteratorOptions
+	bi.it = bi.txn.NewIterator(opts)
+	return &bi
+}
+
 // Stop gracefully stops all disk writes. After calling this, it's safe to kill the process without losing data.
 func (b *BadgerDB) Stop(ctx context.Context) error {
 	return b.backend.Close()
 }
+
+type badgerIterator struct {
+	scope      Scope
+	fullPrefix []byte
+	txn        *badger.Txn
+	it         *badger.Iterator
+}
+
+func (bi *badgerIterator) Close() {
+	bi.it.Close()
+	bi.txn.Discard()
+}
+
+func (bi *badgerIterator) Seek(prefix []byte) {
+	bi.fullPrefix = append(bi.scope.Bytes(), prefix...)
+	bi.it.Seek(bi.fullPrefix)
+}
+
+func (bi *badgerIterator) Next() bool {
+	bi.it.Next()
+	return bi.it.ValidForPrefix(bi.fullPrefix)
+}
+
+func (bi *badgerIterator) Key() []byte {
+	return bi.it.Item().Key()[len(bi.scope.Bytes()):]
+}
+
+func (bi *badgerIterator) Value() []byte {
+	value, _ := bi.it.Item().ValueCopy(nil)
```

Comment:
Let's use `Item().Value()` since we do so in the `Key` method. We can add `KeyCopy` and `ValueCopy` to the iterator later, if we need them.
Also, let's put warnings in the interface about key and value not being copied by default.

Reply:
done